actors/market/src/state.rs

```diff
@@ -583,6 +674,12 @@ impl State {
         let ever_updated = state.last_updated_epoch != EPOCH_UNDEFINED;
         let ever_slashed = state.slash_epoch != EPOCH_UNDEFINED;
 
+        if !ever_updated {
+            // pending deal might have already been removed in first cron_tick, so we don't care if
+            // it's already missing
+            self.remove_pending_deal(store, *deal_cid)?;
```

Comment:
I don't understand this. If the deal has never been updated, there wasn't a cron tick that could have removed it. It could be gone because the deal was manually settled earlier though?

Reply:
on first processing by cron, it simply removes the pending deal. cron can't mark it as updated on that epoch as it doesn't perform any incremental settlement. 

but i'll modify the comment here as yes, it could also have been removed during manual settlement